ansible/roles/grafana/tasks/main.yml

```diff
@@ -0,0 +1,47 @@
+---
+- include_vars: ../../telegraf/defaults/main.yml
+
+- name: configure grafana repo
+  copy: src=grafana.repo dest=/etc/yum.repos.d/grafana.repo owner=root group=root
+
+- name: install grafana
+  yum: name=grafana-{{ grafana.version }} state=present
+
+- name: configure grafana
+  template: src=grafana.ini.j2 dest={{ grafana.conf_dir }}/grafana.ini
+  notify:
+    - restart grafana
+
+- meta: flush_handlers
+
+- name: start grafana-server
+  service: name=grafana-server enabled=yes state=started
+
+- name: install python-httplib2.noarch
+  yum: name=python-httplib2.noarch state=present
+
+- wait_for: port={{ grafana.port }} delay=10 timeout=20
+
+- name: setup upswing datasource in grafana
+  uri:
+    url: http://{{ grafana.host }}:{{ grafana.port }}/api/datasources
```

Comment:
`{{ grafana.host }}` could be rewritten as using `hostvars` and `groups`
ref: http://serverfault.com/questions/638507/how-to-access-host-variable-of-a-different-host-with-ansible